### ISSUE TITLE
[LG] Optional HeroCard as an Activity Attachment

### ISSUE THREAD

**Issue by sschutten** [microsoft/botbuilder-dotnet#3619]
## Version
4.7.2-preview

## Describe the bug
I'm trying to use an optional HeroCard as an Activity Attachment. Based on a condition I want to include the card as an attachment. I throught of using the [pre-built if-function](https://github.com/microsoft/BotBuilder-Samples/blob/master/experimental/common-expression-language/prebuilt-functions.md#logical-comparison-functionsl). It doesn't work if I put the call to the template in either the valueIfTrue or valueIfFalse. If call to the card template is placed in both the valueIfTrue and valueIfFalse the card gets rendered.

## To Reproduce
Below are the templates I use. `Answer` is a string and `prompts` is an array of string values. In my tests `prompts` contains one value, so `empty(prompts)` should be false and the `QnaCardActionsCard(prompts)` should be called.

```
# QnaCard (answer, prompts)
[Activity
    Text = @{answer}
    Attachments = @{if(empty(prompts), {}, QnaCardActionsCard(prompts))}
]

# QnaCardActionsCard (prompts)
[HeroCard
    buttons=@{foreach(prompts, prompt, QnaCardAction(prompt))}
]

# QnaCardAction (prompt)
[CardAction
    Title = @{prompt}
    Value = @{prompt}
    type = imBack
]
```

## Expected behavior
I would expect the card to be included if the condition `empty(prompts)` is false.

**Comment by Danieladu** [comment]
hi sschutten, I just tried to reproduce the issue but did not get the same result as you. 
So, if there are some details that you did not write down in this issue description.

I have commit the test case here, https://github.com/microsoft/botbuilder-dotnet/blob/7e41ee80f9af79800d91044b3ca989075c6254f0/tests/Microsoft.Bot.Builder.LanguageGeneration.Tests/TemplatesTest.cs#L24
You can run this test to check the result.

Another thing, LG has migrate the expression mark from "@{}" to "${}",welcome to have a try.
Thanks.

**Comment by sschutten** [comment]
@Danieladu thanks for investigating. I can confirm it works in 4.8.0-rc0. It appears I was using 4.7.2-preview instead. I've updated the issue description to reflect that. In 4.7.2-preview it wasn't working.

To add to this, I had to change
```
Attachments = @{if(empty(prompts), {}, QnaCardActionsCard(prompts))}
```

to 

```
Attachments = @{if(empty(prompts), [], QnaCardActionsCard(prompts))}
```

to get it to work properly. In 4.7.2-preview I also wasn't able to return an empty array using `[]`, but that seems to be fixed as wel in 4.8.0-rc0. In my first example I returned an empty object. The framework tries to render this as an attachment, but obviously there's no renderer available for an empty object. The empty array solved this.

With the empty object you get this message in the emulator:
![image](https://user-images.githubusercontent.com/10097564/77645914-de491b80-6f63-11ea-91d1-47cb814ba3f5.png)
